src/Jira/Api.php

```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
 		return $result;
 	}
 
+	 /**
+	 * Returns project components.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array|false
+	 * @since  2.0.0
+	 */
+	public function getProjectComponents($project_key)
+	{
+		return $this->api(self::REQUEST_GET, sprintf('/rest/api/2/project/%s/components', $project_key), array(), true);
+	}
+
+	 /**
+	 * Get all issue types with valid status values for a project.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array
+	 * @since  2.0.0
+	 */
+	public function getAllIssueTypes($project_key) 
```

Comment:
We already have a getIssueTypes who returns the global list. This one is specific for projects. Maybe rename to getProjectIssueTypes?